.github/workflows/test-e2e.yml

```diff
@@ -1,9 +1,53 @@
 name: ocelot.social end-to-end test CI
-on: push
+on:
+  push:
+    branches:
+      - cache-docker-images-in-e2e-github-flow
 
 jobs:
+  docker_preparation:
+    name: Fullstack test preparation
+    runs-on: ubuntu-latest
+    outputs:
+      cache-hit: ${{ steps.cache.outputs.cache-hit }}
+      pr-number: ${{ steps.pr.outputs.number }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Copy env files
+        run: |
+          cp webapp/.env.template webapp/.env
+          cp backend/.env.template backend/.env
+
+      - name: Build docker images
+        run: |
+          mkdir /tmp/images
+          docker build --target community -t "ocelotsocialnetwork/neo4j-community:test" neo4j/
+          docker save "ocelotsocialnetwork/neo4j-community:test" > /tmp/images/neo4j.tar
+          docker build --target test -t "ocelotsocialnetwork/backend:test" backend/
+          docker save "ocelotsocialnetwork/backend:test" > /tmp/images/backend.tar
+          docker build --target test -t "ocelotsocialnetwork/webapp:test" webapp/
+          docker save "ocelotsocialnetwork/webapp:test" > /tmp/images/webapp.tar
+      
+      - name: Get pr number
+        id: pr
+        uses: 8BitJonny/gh-get-current-pr@2.2.0
+      
+      - name: Cache docker images
+        id: cache
+        uses: actions/cache@v3.2.6
+        with:
+          path: /tmp/images/
+          key: e2e-preparation-cache-pr${{ steps.pr.outputs.number }}
```

Comment:
maybe you want to use the github action run instead to avoid cash misses? Not sure thats the reason for the full stack test currently failing, just a wild guess? 

![image](https://github.com/Ocelot-Social-Community/Ocelot-Social/assets/1238238/0093c9b5-9d91-4a17-b753-3ef8b514bf04)

`GITHUB_RUN_NUMBER` or `GITHUB_RUN_ID` https://docs.github.com/en/actions/learn-github-actions/variables